doc/cpd/101.rst

```diff
@@ -0,0 +1,150 @@
+CPD-101: Key Storage Encryption
+===============================
+
+Metadata
+--------
+
+* CPD Version: 1
+* Status: Open
+
+Description
+-----------
+Today we are holding keys the same way that secrets are used in some container managers.
+Instead of holding keys in base64 and assuming that the ``Storage`` instance is used
+only for Commissaire, we could encrypt the keys to add another layer of safety.
+
+Rationale
+---------
+The likelihood of having a ``Storage`` system that is used only for Commissaire seems
+low. More than likely the same instance will be used for other applications as well.
+By adding encryption to the keys we mitigate could mitigate access from those with
+access to data dumps and storage system access.
+
+
+Design
+------
+The server(s) would house a GnuPG configuration and keys allowing them to encrypt
+and decrypt data. The servers would either need to have the same key across all
+instances or each server would need a copy of other services public keys.
+
+
+Additional Libraries
+~~~~~~~~~~~~~~~~~~~~
+To facilitate easy access to GnuPG, `python-gnupg <http://pythonhosted.org/gnupg/>`_ would
+need to be installed. This library utilizes ``gpg`` directly so it would need to be
+available as well.
+
+
+HTTP Server Updates
+~~~~~~~~~~~~~~~~~~~
+While ``StorageService`` seems like it would be the proper place to add the
+encryption/decryption it's important to remember the key would then be transfered
+back over the wire before it made it to the http server and, ultimately, back to
+the user. Instead, adding the encryption/decryption at the REST level would keep
+the key encrypted as long as possible.
+
+The HTTP Server would need to handle decrypting the key back into a base64 encoded
+string before handing it back to the authenticated requestor.
```

Comment:
Kombu supports SSL/TLS at the `Connection` layer for *some* transports:
```
SSL compatibility

SSL currently only works with the py-amqp, amqplib, and qpid transports.
For other transports you can use stunnel.
```
Perhaps we could pass SSL/TLS options for `kombu.Connection` from a config file?

[source](http://docs.celeryproject.org/projects/kombu/en/latest/reference/kombu.connection.html)